lib/AST/Type.cpp

```diff
@@ -3631,6 +3631,9 @@ static CachedProperties computeCachedProperties(const Type *T) {
     return Cache::get(cast<PipeType>(T)->getElementType());
   case Type::TypeVariable:
     return CachedProperties(ExternalLinkage, false);
+  case Type::Existential:
```

Comment:
I believe you should use the cached properties of the body of the existential type.   This should be treated the same way as other constructed types such as array types and function types are constructed.

Reply:
Done, but how can we test this?

Reply:
This code still isn't looking at the body of the existential type.  I believe you would need to define a variable with an existential type in one compilation unit, declare it another, and make sure that they link.

Reply:
Done. Added a TODO for the test.